docs/tsp/typeServerProtocol.ts

```diff
@@ -0,0 +1,383 @@
+        Generic = 1 << 6, // Indicates if the type is a generic type (a type that can be parameterized with other types).
+    }
+
+    // Flags that describe the characteristics of a function or method.
+    // These flags can be combined using bitwise operations.
+    export const enum FunctionFlags {
+        None = 0,
+        Async = 1 << 0, // Indicates if the function is asynchronous.
+        Generator = 1 << 1, // Indicates if the function is a generator (can yield values).
+        Abstract = 1 << 2, // Indicates if the function is abstract (must be implemented in a subclass).
+        Static = 1 << 3, // Indicates if the function is static (belongs to the class rather than an instance).
+    }
+
+    // Flags that describe the characteristics of a class.
+    // These flags can be combined using bitwise operations.
+    export const enum ClassFlags {
+        None = 0,
+        Enum = 1 << 0, // Indicates if the class is an enum (a special kind of class that defines a set of named values).
+    }
+
+    export interface Type {
+        // Unique identifier for the type definition within the session.
+        handle: string | number;
+
+        // Essential classification of the symbol.
+        category: TypeCategory;
+
+        // Flags describing the type.
+        flags: TypeFlags;
+
+        // Name of the module the type comes from
+        moduleName: string | undefined;
+
+        // Name of the type.
+        name: string;
+
+        // Flags specific to the category
+        categoryFlags: number;
+    }
+
+    export interface Member {
+        // The name of the member.
+        name: string;
+
+        // The type of the member.
+        type: Type;
+
+        // The type the member came from
+        classType: Type | undefined;
+    }
+
+    // Flags that are used for searching for members of a type.
+    export const enum MemberAccessFlags {
+        None = 0,
+        SkipInstanceMembers = 1 << 0, // Skip instance members when searching for members of a type.
+        SkipObjectBaseClass = 1 << 1, // Skip members from the `object` base class when searching for members of a type.
+    }
+
+    // Represents the category of a declaration in the type system.
+    // This is used to classify declarations such as variables, functions, classes, etc.
+    export const enum DeclarationCategory {
+        Intrinsic = 0, // An intrinsic is a built-in type or function that is part of the language itself, such as `int`, `str`, `len`, etc.
+        Variable = 1, // A variable is a named storage location that can hold a value.
+        Param = 2, // A parameter is a variable that is passed to a function or method.
+        TypeParam = 3, // A type parameter is a placeholder for a type that can be specified when the type is used, such as in generics.
+        TypeAlias = 4, // A type alias is a name that refers to another type, such as `List[int]` or `Dict[str, int]`.
```

Comment:
This is specifically for PEP 695 type aliases. It doesn't apply to older type aliases, which use variable declarations.

Reply:
I'll change the comment to mention that. Or are you suggesting we can drop this as it's not used much?